src/proj.h

```diff
@@ -328,6 +340,17 @@ extern const PJ      *proj_shutdown;
 #endif
 
 
+/* Set or read error level */
+int  proj_errno (PJ *P);
+void proj_errno_set (PJ *P, int err);
```

Comment:
> I much agree that fighting for slim APIs is the good fight

Pleased to see we are aligned on this objective

> I just think my real world experience differ from yours wrt. error reporting: I have always found it very useful to be able to set an error level on a state object from user/application 

OK, if this is on purpose, let's go for it

> Regarding the lack of a proj internal include file: Wasn't the idea of introducing proj_api.hto demote projects.h to "internal include" status?

Perhaps, but the issue is that projects.h was already a installed header (since it is include_HEADERS in https://github.com/OSGeo/proj.4/blob/master/src/Makefile.am#L9), so there's always a temptation from API users to use things even if marked as internal details. I really thing using a proj_internal.h that is not installed is the clean way of keeping internal details hidden reliably